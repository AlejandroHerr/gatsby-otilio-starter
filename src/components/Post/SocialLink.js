import React from 'react';
import PropTypes from 'prop-types';

import SimpleIcon from '../icons/SimpleIcon';

import styles from './Post.module.scss';

const SocialLink = ({ service, url }) => (
  <a className={styles.post_author__social} href={url}>
    <SimpleIcon brand={service.toLowerCase()} className={styles.post_author__social_icon} />
  </a>
);

SocialLink.propTypes = {
  service: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default SocialLink;
