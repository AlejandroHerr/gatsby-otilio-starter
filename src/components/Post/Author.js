// @flow
import React from 'react';
import Img from 'gatsby-image';

import type { AuthorType } from '../../types/post';

import ByteIcon from '../icons/ByteIcon';
import SocialLink from './SocialLink';

import styles from './Post.module.scss';

const Author = ({ author }: {author: AuthorType}) => (
  <aside className={styles.post_author}>
    <figure className={`${styles.post_author__avatar} ${styles.avatar}`}>
      <Img sizes={author.avatar.childImageSharp.sizes} />
    </figure>
    <div className={styles.post_author__bio}>
      <h4 className={styles.post_author__name}>{author.name}</h4>
      <p className={styles.post_author__about}>{author.about}</p>
      <span className={styles.post_author__location}>
        <ByteIcon className={styles.post_author__location_icon} icon="location" /> {author.location}
      </span>
      {author.social.map(({ service, url }) => (
        <SocialLink key={service} service={service} url={url} />
      ))}
    </div>
  </aside>
);

export default Author;
