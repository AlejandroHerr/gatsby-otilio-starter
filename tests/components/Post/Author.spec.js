import React from 'react';
import { shallow } from 'enzyme';

import Img from 'gatsby-image';

import Author from '../../../src/components/Post/Author';
import SocialLink from '../../../src/components/Post/SocialLink';

const defaultProps = {
  author: {
    name: 'Test Name',
    about: 'Lorem Ipsum',
    avatar: {
      childImageSharp: {
        sizes: {
          aspectRatio: 1,
          sizes: 'sizes',
          src: 'src',
          srcSet: 'srcSet',
        },
      },
    },
    location: 'Test Location',
    social: [
      { service: 'Test Service 1', url: 'Url 1' },
      { service: 'Test Service 2', url: 'Url 2' },
    ],
  },
};

const setup = (props = {}) => {
  const author = shallow(<Author {...defaultProps} {...props} />);

  return {
    author,
    image: author.find(Img),
    socialLinks: author.find(SocialLink),
  };
};

describe('Post', () => {
  describe('Author', () => {
    it('should render a Gatsby Image', () => {
      const { author, image } = setup();

      expect(image).toHaveLength(1);
      expect(image.prop('sizes')).toBe(defaultProps.author.avatar.childImageSharp.sizes);
      expect(author).toMatchSnapshot();
    });
    it('should render SimpleIcons for every social icon', () => {
      const { author, socialLinks } = setup();

      expect(socialLinks).toHaveLength(defaultProps.author.social.length);

      defaultProps.author.social.forEach((link, idx) => {
        const socialLink = socialLinks.at(idx);

        expect(socialLink.prop('service')).toBe(link.service);
        expect(socialLink.prop('url')).toBe(link.url);
      });
      expect(author).toMatchSnapshot();
    });
  });
});
