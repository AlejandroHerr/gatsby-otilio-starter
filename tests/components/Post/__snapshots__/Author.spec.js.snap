// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Author should render SimpleIcons for every social icon 1`] = `
<aside
  className="post_author"
>
  <figure
    className="post_author__avatar"
  >
    <Image
      Tag="div"
      alt=""
      fadeIn={true}
      sizes={
        Object {
          "aspectRatio": 1,
          "sizes": "sizes",
          "src": "src",
          "srcSet": "srcSet",
        }
      }
    />
  </figure>
  <div
    className="post_author__bio"
  >
    <h4
      className="post_author__name"
    >
      Test Name
    </h4>
    <p
      className="post_author__about"
    >
      Lorem Ipsum
    </p>
    <span
      className="post_author__location"
    >
      <ByteIcon
        bold={false}
        className="post_author__location_icon"
        icon="location"
      />
       
      Test Location
    </span>
    <SocialLink
      key="Test Service 1"
      service="Test Service 1"
      url="Url 1"
    />
    <SocialLink
      key="Test Service 2"
      service="Test Service 2"
      url="Url 2"
    />
  </div>
</aside>
`;

exports[`Post Author should render a Gatsby Image 1`] = `
<aside
  className="post_author"
>
  <figure
    className="post_author__avatar"
  >
    <Image
      Tag="div"
      alt=""
      fadeIn={true}
      sizes={
        Object {
          "aspectRatio": 1,
          "sizes": "sizes",
          "src": "src",
          "srcSet": "srcSet",
        }
      }
    />
  </figure>
  <div
    className="post_author__bio"
  >
    <h4
      className="post_author__name"
    >
      Test Name
    </h4>
    <p
      className="post_author__about"
    >
      Lorem Ipsum
    </p>
    <span
      className="post_author__location"
    >
      <ByteIcon
        bold={false}
        className="post_author__location_icon"
        icon="location"
      />
       
      Test Location
    </span>
    <SocialLink
      key="Test Service 1"
      service="Test Service 1"
      url="Url 1"
    />
    <SocialLink
      key="Test Service 2"
      service="Test Service 2"
      url="Url 2"
    />
  </div>
</aside>
`;
